### PR TITLE
Store creation M3: profiler question - optional category selector

### PR DESCRIPTION
**Pull request by jaclync**
Part of #8376 

### Why

In order for WooCommerce to provide better customizations for the new store after store creation, we want to add a few questions to the store creation flow to understand more about the store profile. The first question is about the store category, which indicates the store's industry. This question is optional and can be skipped by tapping the `Continue` CTA without selecting an option or by tapping the `Skip` CTA.

### Changes

This PR only includes the UI part, as the API is still WIP (p1670825052838929/1670823717.177509-slack-C01SFMVEYAK). The profiler question is only shown behind a new feature flag `storeCreationM3Profiler`, and the selected category is shown in the store summary screen.

In order to share code in the 3 profiler questions (design in HyVloP5FipZzyPVenH2euI-fi-3066%3A144169&t=rYz8R7hZHuYF1IFo-0) where two are optional and one is required, I created a generic `StoreCreationProfilerQuestionView` for just the main content view that can take in any question content along with shared UI that includes 3 labels at the top. Then, there's  `OptionalStoreCreationProfilerQuestionView` that wraps `StoreCreationProfilerQuestionView` with 2 navigation CTAs - the `Continue` and `Skip` CTAs. When it's time to implement the required question https://github.com/woocommerce/woocommerce-ios/issues/8378, I plan to create another wrapper `RequiredStoreCreationProfilerQuestionView` with only a `Continue` CTA with an observable boolean for whether the CTA is enabled.

`StoreCreationCategoryQuestionView` is the final category question view with a view model `StoreCreationCategoryQuestionViewModel` that implements the view model protocol for both `StoreCreationProfilerQuestionViewModel` and `OptionalStoreCreationProfilerQuestionViewModel`. Any suggestions on this implementation are welcome!

#### Feature flag enabled

- Log in if needed
- Go to the Menu tab, and tap `Switch store`
- On the store picker, tap `+ Add a store`
- Tap `Create a new store` --> the store name form should be shown
- Enter a store name and continue --> the category question should be shown with a `Skip` CTA at the top and a `Continue` CTA at the bottom
- Tap `Skip` --> the domain selector should be shown
- Go back to the previous screen
- Select an option and tap `Continue` --> the domain selector should be shown
- Select a domain and continue --> the selected category name should be displayed on the store summary below the URL

#### Feature flag disabled

- In code, return `false` manually for `storeCreationM3Profiler` feature flag in `DefaultFeatureFlagService`
- Log in if needed
- Go to the Menu tab, and tap `Switch store`
- On the store picker, tap `+ Add a store`
- Tap `Create a new store` --> the store name form should be shown
- Enter a store name and continue --> the domain selector should be shown
- Select a domain and continue --> there should be no store category below the URL on the store summary

### Screenshots

dark | light
-- | --
Simulator Screen Shot - iPhone 14 Pro - 2022-12-12 at 14 52 41 | Simulator Screen Shot - iPhone 14 Pro - 2022-12-12 at 14 52 47
Simulator Screen Shot - iPhone 14 Pro - 2022-12-12 at 14 51 24 | Simulator Screen Shot - iPhone 14 Pro - 2022-12-12 at 14 51 04

Screencast:

https://user-images.githubusercontent.com/1945542/206979298-1ceaebd3-25d0-4bad-881b-5b1bb003a8f2.mov

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Authentication/Store Creation/Profiler/Category/StoreCreationCategoryQuestionView.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    /// Shows a transparent navigation bar without a bottom border and with a close button to dismiss.
+    func configureNavigationBarAppearance() {
+        let appearance = UINavigationBarAppearance()
+        appearance.configureWithTransparentBackground()
+        appearance.backgroundColor = .systemBackground
+
+        navigationItem.standardAppearance = appearance
+        navigationItem.scrollEdgeAppearance = appearance
+        navigationItem.compactAppearance = appearance
+    }
```
Comment (itsmeichigo) [inline WooCommerce/Classes/Authentication/Store Creation/Profiler/Category/StoreCreationCategoryQuestionView.swift]: I see that we're using this style a lot so I added an extension for this [here](https://github.com/woocommerce/woocommerce-ios/blob/69be467b056c9ef4bb7cd3c5c2220970880fc4db/WooCommerce/Classes/Extensions/UIViewController%2BHelpers.swift#L38). Maybe we should use it instead?

File: WooCommerce/Classes/Authentication/Store Creation/Profiler/Category/StoreCreationCategoryQuestionViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        .init(name: NSLocalizedString("Food and Drink",
+                                      comment: "Option in the store creation category question."),
+              value: ""),
+        .init(name: NSLocalizedString("Arts and Crafts",
+                                      comment: "Option in the store creation category question."),
+              value: ""),
+        .init(name: NSLocalizedString("Health and Beauty",
+                                      comment: "Option in the store creation category question."),
+              value: ""),
+        .init(name: NSLocalizedString("Pets Pet Care",
```
Comment (itsmeichigo) [inline WooCommerce/Classes/Authentication/Store Creation/Profiler/Category/StoreCreationCategoryQuestionViewModel.swift:51]: Is this a typo? 😅

**Review by itsmeichigo — APPROVED** [review]
Overall this looks good to me! I spot a few minor differences with the design, though:

- The category name on the profiler screen is not capitalized like on the design.
- The title "Site Category" is not displayed when the list is scrolled up. I'm not sure if it's possible to handle it, given that the big title is different.

| Design | App |
| ----- | ----- |
|  |  |
|  |  |

On an unrelated note, I realized that I could drag to dismiss on the summary screen, and there was no dialog to confirm before the view was dismissed. This may be troublesome because I don't think there's an entry point to continue payment after dismissing the flow. Please ignore this if there has already been a subtask for this.

File: WooCommerce/Classes/Authentication/Store Creation/Profiler/Category/StoreCreationCategoryQuestionView.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/Authentication/Store Creation/Profiler/Category/StoreCreationCategoryQuestionView.swift]: Yup thanks for creating that extension, updated in https://github.com/woocommerce/woocommerce-ios/pull/8379/commits/87be413bba013e4b19549c0f75fdbe798ab77ea3

**Comment by jaclync** [comment]
> * The category name on the profiler screen is not capitalized like on the design.

Great catch, updated in https://github.com/woocommerce/woocommerce-ios/pull/8379/commits/7142d88934f44581bab08048840095d9cfa1a302

> * The title "Site Category" is not displayed when the list is scrolled up. I'm not sure if it's possible to handle it, given that the big title is different.

As you said, this isn't easily feasible since we aren't using large title as there is another label above the title. I clarified with Ann about this earlier this week, and the design was updated in HyVloP5FipZzyPVenH2euI-fi-3760%3A138977&t=gO7KflHr1ccRB4nZ-0

> On an unrelated note, I realized that I could drag to dismiss on the summary screen, and there was no dialog to confirm before the view was dismissed. This may be troublesome because I don't think there's an entry point to continue payment after dismissing the flow. Please ignore this if there has already been a subtask for this.

That's a good observation & suggestion, I disabled the interactive dismissal in https://github.com/woocommerce/woocommerce-ios/pull/8379/commits/8a9ce6b7406d919ec8ae3f9af6ed0f9569c8a37c. The user can still go back to the initial screen and dismiss the screen where a confirmation alert is shown. We could also re-enable interactive dismissal and show a confirmation alert later.

---

Thanks for your feedback, I'll merge this when CI passes to make development easier for other profiler questions.

File: WooCommerce/Classes/Authentication/Store Creation/Profiler/Category/StoreCreationCategoryQuestionViewModel.swift (hunk 2 above)
Comment (jaclync) [reply WooCommerce/Classes/Authentication/Store Creation/Profiler/Category/StoreCreationCategoryQuestionViewModel.swift:51]: This was in the design and I had the same question, but we're also updating the list and separating it into two sections as the latest design after syncing with the Ghidorah team. There's a subtask to update the design in https://github.com/woocommerce/woocommerce-ios/issues/8376, where the categories will be updated

**Pull request merged**